Makefile

```
@@ -364,55 +429,108 @@ $(OBJECT_DIR)/$(TARGET)/%.o: %.S | $$(dir $$@)
+
+ifdef IN_SUMMARY_OUTER
+# Outer intercepted invocation: real work happens in the first goal's recipe.
+# Remaining goals are @: no-ops so make doesn't error on unknown targets.
+# summary_build exits non-zero on any failure, which causes the first goal's
+# recipe to fail — make reliably propagates a goal's own recipe failure to
+# the outer exit code (unlike prerequisite failure propagation in Make 4.4+).
+# _INVALID_GOALS are pre-populated into the failed marker so they appear in
+# the summary count even though they produce no marker entry themselves.
+_INVALID_GOALS := $(filter-out $(VALID_TARGETS), $(MAKECMDGOALS))
+.PHONY: $(MAKECMDGOALS)
+
```

checkmake (0.3.2)

[warning] Required target "all" is missing from the Makefile.

(minphony)